libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVp8Reader.java

```diff
@@ -195,6 +200,23 @@ private boolean validateVp8Descriptor(ParsableByteArray payload, int packetSeque
     return true;
   }
 
+  /**
+   * Outputs sample metadata.
+   *
+   * <p>Call this method only when receiving a end of VP8 partition
+   */
+  private void outputSampleMetadataForFragmentedPackets() {
+    trackOutput.sampleMetadata(
+        sampleTimeUsOfFragmentedSample,
+        isKeyFrame ? C.BUFFER_FLAG_KEY_FRAME : 0,
+        fragmentedSampleSizeBytes,
+        /* offset= */ 0,
+        /* cryptoData= */ null);
+    fragmentedSampleSizeBytes = 0;
```

Comment:
It seems if `fragmentedSampleSizeBytes` is reset to zero here, the logic on L100 will run on 

- The first received RTP packet, or
- The first received RTP packet after a seek.

Is that correct? If so, is it intentional?

Reply:
Like I think what you did in #115 looks "more correct"?

Reply:
I was asking a yes-or-no question here - the logic 

```java
      if (fragmentedSampleSizeBytes == C.LENGTH_UNSET && gotFirstPacketOfVp8Frame) {
        isKeyFrame = (data.peekUnsignedByte() & 0x01) == 0;
      }
```

runs on the first received RTP packet, or the first packet arrives after a seek ONLY, because you reset it to 0 in `outputSampleMetadataForFragmentedPackets` rather than `LENGTH_UNSET`.

**Is that correct? If so, is it intentional?**

Reply:
It was not intentional, fragmentedSampleSizeBytes should have been reset with LENGTH_UNSET. It should run on every first received RTP packet of a fragment.

Reply:
Thanks for the clarification.